docs/notes/02-packaging.md

```diff
@@ -48,6 +49,106 @@ testing notes](05-linters-and-tests.md) for more info.
+     2. [mkdocs_section_index](https://oprypin.github.io/mkdocs-section-index/): A MkDocs plugin that adds a navigation section index to the documentation sidebar. It enhances navigation by providing an index of the sections in the documentation.
+     3. [mkdocs_gen_files](https://oprypin.github.io/mkdocs-gen-files/): A MkDocs plugin that allows the generation of additional files during the documentation build process. It provides functionality to dynamically generate files that can be included in the documentation.
+     4. [mkdocs_literate_nav](https://oprypin.github.io/mkdocs-literate-nav/): A MkDocs plugin that enhances the navigation sidebar by providing collapsible sections. It improves the readability and organization of the documentation by allowing users to collapse and expand sections.
+     5. [mkdocs-gallery](https://smarie.github.io/mkdocs-gallery/): A MkDocs plugin that enables the creation of image galleries in the documentation. It provides functionality for creating and displaying image galleries with customizable options.
+     6. [pillow](https://pillow.readthedocs.io/en/stable/): The Python Imaging Library (PIL) fork known as Pillow. It is used for image processing tasks, such as resizing, cropping, and modifying images, which can be utilized in the documentation.
+
+   - `dev`: Developer dependencies.
+     1. [black](https://black.readthedocs.io/en/latest/): Code formatter. It automatically formats your Python code according to the Black code style.
+     2. [isort](https://isort.readthedocs.io/en/latest/): Import sorter. It automatically organizes and sorts import statements in your Python code.
+     3. [pip-tools](https://pip-tools.readthedocs.io/en/latest/): Dependency management. It helps manage and synchronize project dependencies, allowing you to generate and maintain requirements.txt or pipfile.lock files.
+     4. [pytest](https://docs.pytest.org/en/7.3.x/): Testing framework. It is a popular testing framework for Python that allows you to write and execute tests easily.
+     5. [flake8](https://flake8.pycqa.org/en/latest/): Code linter. It checks your Python code for style, potential errors, and adherence to coding conventions.
+     6. [pytest-cov](https://pytest-cov.readthedocs.io/en/latest/): Test coverage plugin for pytest. It integrates test coverage measurement into the pytest framework, providing coverage reports and analysis for your tests.
+
+    **Syntax example:**
+    ```toml
+      [project.optional-dependencies]
+      docs = [
+        'mkdocs',
+        'mkdocstrings[python]',
+        ...
+      ]
+      dev = [
+        "black", 
+        "isort",                       
+        ...
+      ]
+    ```
+
+
+3. `[tool.setuptools.packages.find]`: configure the package discovery behavior for setuptools. The find function of setuptools locates and returns the packages within a Python project. You can provide the following details:
```

Comment:
I don't think we're using this? this is a way of automatically determining what your requirements are, right? whereas we are listing them epxlicitly, and so this isn't necessary. Or am I misunderstanding what this does?

Reply:
also, the docs say this is a beta feature, so I don't think we should use it

Reply:
I checked and the pip install works on my linux mahcine with python3.10 if i just remove the setup tools find. 
we should keep an eye on it, because in a non-flat repo structure it used to create problem and not finding the source code